File: server/src/main/java/io/github/hulang1024/chinesechessserver/play/rule/ChessHost.java
package io.github.hulang1024.chinesechessserver.domain.chinesechess.rule;

/**
 * 棋方
 */
public enum ChessHost {
    RED(1),
    BLACK(2);
    
    private int code;

    ChessHost(int code) {
        this.code = code;
    }

    public int code() {
        return this.code;
    }

    public ChessHost reverse() {
        return this == RED ? BLACK : RED;
    }

    public static ChessHost fromCode(int code) {
        return code == 1 ? RED : BLACK;
    }
}

File: client/src/rule/chess_host.ts
/**
 * 棋方
 */
enum ChessHost {
    /**
     * 先手
     */
    RED = 1,

    /**
     * 后手
     */
    BLACK = 2
}

export default ChessHost;

// eslint-disable-next-line
namespace ChessHost {
  // eslint-disable-next-line
  export function reverse(host: ChessHost) {
    return host == ChessHost.BLACK ? ChessHost.RED : ChessHost.BLACK;
  }
}
